### PR TITLE
Added kind support for README.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Added kind support for clusterctl README.
```

### CODE REVIEW

File: cmd/clusterctl/README.md [hunk 1]
```diff
@@ -54,6 +54,12 @@ Additional advanced flags can be found via help.
 
 ### Interacting with your cluster
 
+If you are using kind, config the `KUBECONFIG` first before using kubectl:
```
Comment (detiber) [inline cmd/clusterctl/README.md]:
```suggestion
If you are using kind, set the `KUBECONFIG` environment variable first before using kubectl:
```

File: cmd/clusterctl/README.md [hunk 2]
```diff
@@ -13,8 +13,8 @@ this repository.**
 
 ### Prerequisites
 
-1. Install [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/)
-2. Install a [driver](https://github.com/kubernetes/minikube/blob/master/docs/drivers.md) for minikube. For Linux, we recommend kvm2. For MacOS, we recommend VirtualBox.
+1. Install [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) or [kind](https://github.com/kubernetes-sigs/kind#installation-and-usage).
```
Comment (vincepri) [inline cmd/clusterctl/README.md]: Should we prefer `kind` here and move the minikube references in another optional section?

Comment (gyliu513) [reply cmd/clusterctl/README.md]: I put `kind` in the first and also mentioned `kind` as preferred, hope it is OK.

File: cmd/clusterctl/README.md (hunk 1 above)
Comment (gyliu513) [reply cmd/clusterctl/README.md]: Done.

File: cmd/clusterctl/README.md [hunk 3]
```diff
@@ -41,10 +41,12 @@ https://github.com/kubernetes-sigs/cluster-api/issues/158 and https://github.com
 1. Create a cluster:
 
    ```shell
-   ./clusterctl create cluster --provider <provider> -c cluster.yaml -m machines.yaml -p provider-components.yaml -a addons.yaml
+   ./clusterctl create cluster --provider <provider> --bootstrap-type <bootstrap-type> -c cluster.yaml -m machines.yaml -p provider-components.yaml -a addons.yaml
    ```
 
-To choose a specific minikube driver, please use the `--vm-driver` command line parameter. For example to use the kvm2 driver with clusterctl you would add `--vm-driver kvm2`
+We support two `bootstrap-type` now, including kind and minikube.
```
Comment (neolit123) [inline cmd/clusterctl/README.md]: > Currently two `bootstrap-type` options are supported - `kind` and `minikube`.

Comment (gyliu513) [reply cmd/clusterctl/README.md]: Done.

File: cmd/clusterctl/README.md [hunk 4]
```diff
@@ -41,10 +41,12 @@ https://github.com/kubernetes-sigs/cluster-api/issues/158 and https://github.com
...
 
    ```shell
-   ./clusterctl create cluster --provider <provider> -c cluster.yaml -m machines.yaml -p provider-components.yaml -a addons.yaml
+   ./clusterctl create cluster --provider <provider> --bootstrap-type <bootstrap-type> -c cluster.yaml -m machines.yaml -p provider-components.yaml -a addons.yaml
    ```
 
-To choose a specific minikube driver, please use the `--vm-driver` command line parameter. For example to use the kvm2 driver with clusterctl you would add `--vm-driver kvm2`
+Currently two `bootstrap-type` options are supported - `kind` and `minikube`.
+
+If you are using minikube, to choose a specific minikube driver, please use the `--vm-driver` command line parameter. For example to use the kvm2 driver with clusterctl you would add `--vm-driver kvm2`
```
Comment (detiber) [inline cmd/clusterctl/README.md]: I believe this needs to be passed through using `--bootstrap-flags` instead of passed directly.

Comment (gyliu513) [reply cmd/clusterctl/README.md]: @detiber it is being fixed at https://github.com/kubernetes-sigs/cluster-api/pull/846

**Comment by gyliu513** [comment]
@detiber rebased, can you please help check?

**Pull request merged**